java/code/src/com/redhat/rhn/manager/satellite/StartupTasksCommand.java

```diff
@@ -0,0 +1,56 @@
+/**
+ * Copyright (c) 2020 SUSE LLC
+ *
+ * This software is licensed to you under the GNU General Public License,
+ * version 2 (GPLv2). There is NO WARRANTY for this software, express or
+ * implied, including the implied warranties of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. You should have received a copy of GPLv2
+ * along with this software; if not, see
+ * http://www.gnu.org/licenses/old-licenses/gpl-2.0.txt.
+ *
+ * Red Hat trademarks are not licensed under GPLv2. No permission is
+ * granted to use or replicate Red Hat trademarks that are incorporated
+ * in this software or its documentation.
+ */
+
+package com.redhat.rhn.manager.satellite;
+
+import com.redhat.rhn.common.hibernate.HibernateFactory;
+import com.redhat.rhn.domain.contentmgmt.ContentProjectFactory;
+import com.redhat.rhn.manager.BaseTransactionCommand;
+
+import org.apache.log4j.Logger;
+
+/**
+ * Tasks to be performed on the application startup
+ */
+public class StartupTasksCommand extends BaseTransactionCommand {
```

Comment:
Most cleanups of this kind currently take place in Taskomatic. What would be the pros and cons in this specific case?

Reply:
Handling the problem in Tasko:

Pros: staying consistent with the most cleanup tasks ;)

Cons: Taskomatic doesn't know which tasks are stale, so it' impossible to flag those as failed: The tasks are run as asynchronous actions in tomcat. Each task has an attribute `status` (`BUILDING`, `FAILED`, `COMPLETED`,...). When the CLM build logic runs (in tomcat), the `status` is set based on the progress of the build. Taskomatic has no way to figure out if a task in `BUILDING` state is still doing something or if it's stale. OTOH, on Tomcat startup, we can safely say that all `BUILDING` tasks are stale.

Reply:
Then I understand that a Taskomatic based approach has no chance.

You have my blessing for the approach.